src/thor/bidirectional_astar.cc

```diff
@@ -297,7 +311,10 @@ inline bool BidirectionalAStar::ExpandForwardInner(GraphReader& graphreader,
   if (expansion_callback_) {
     expansion_callback_(graphreader, "bidirectional_astar", meta.edge_id, "r", false);
   }
-  return true;
+
+  // we've just added this edge to the queue, but we won't expand from it if it's a not-thru edge that
+  // will be pruned. In that case we want to allow uturns.
+  return !(pred.not_thru_pruning() && meta.edge->not_thru());
```

Comment:
This change handles the case where we continue to enqueue not_thru edges like we did before, but we return false _if_ we're about to prune this edge. This allows the calling function to enqueue a u-turn edge in addition to the not_thru edge.